tests/test_agents.py

```diff
@@ -1061,6 +1061,25 @@ async def test_clinical_tool_usage(agent_test_settings) -> None:
     ), "No clinical trials were put into contexts"
 
 
+@pytest.mark.asyncio
+async def test_search_pagination(agent_test_settings: Settings) -> None:
+    """Test that pagination works correctly in SearchIndex.query()."""
+    index = await get_directory_index(settings=agent_test_settings)
```

Comment:
One other comment is, this index build in CI is a bit slow.

Can you do this:

1. Make an async `pytest.fixture` with `session` scope that does this
2. Use it in `test_search_pagination`, `test_deepcopy_env`

Then this PR is good to go

Reply:
Per in-person convo, `agent_test_settings` being not session scoped messes this up.

It's too much work to undo this for now, so we can handle this in a separate PR. YAGNI